### PR TITLE
Add CatchpointTracking to config file

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

Add a CatchpointTracking option to the config file, so that non-archival nodes would not track catchpoints by defaults.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -328,6 +328,13 @@ type Local struct {
 	// OptimizeAccountsDatabaseOnStartup controls whether the accounts database would be optimized
 	// on algod startup.
 	OptimizeAccountsDatabaseOnStartup bool `version[10]:"false"`
+
+	// CatchpointTracking determines if catchpoints are going to be tracked. The value is interpreted as follows:
+	// A of -1 means "don't track catchpoints".
```
Comment (algonautshant) [inline config/config.go:333]: type: A of -> A value of

File: config/config.go [hunk 2]
```diff
@@ -328,6 +328,13 @@ type Local struct {
...
+	// A value of 1 means "track catchpoints as long as CatchpointInterval is also set to a positive non-zero value". If CatchpointInterval <= 0, no catchpoint tracking would be performed.
```
Comment (algonautshant) [inline config/config.go:334]: Should there be a comment on CatchpointInterval that it's effect may be overruled by CatchpointTracking?

File: config/config.go (hunk 1 above)
Comment (tsachiherman) [reply config/config.go:333]: correct. I will fix it in the next pr ( just so it won't delay the merge too much )

File: config/config.go (hunk 2 above)
Comment (tsachiherman) [reply config/config.go:334]: I can't see a good reason not to do so. Will do on next PR.

**Pull request merged**